OADP-3399: Reworking OpenShift KubeVirt OADP docs (#72372)
openshift/openshift-docs: backup_and_restore/application_backup_and_restore/installing/installing-oadp-kubevirt.adoc

include::_attributes/common-attributes.adoc[]
:context: installing-oadp-kubevirt
:installing-oadp-kubevirt:
:credentials: cloud-credentials
:provider: kubevirt

toc::[]

You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.

// include::snippets/oadp-mtc-operator.adoc[]

Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[{oadp-full}].

[NOTE]
====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups

* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.

* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====

include::modules/install-and-configure-oadp-kubevirt.adoc[leveloffset=+1]

> 
> // You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.

Author: ```suggestion
[role="_additional-resources"]
.Additional resources

. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
.  xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

:_mod-docs-content-type: ASSEMBLY
[id="installing-oadp-kubevirt"]
// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"
= Configuring the {oadp-full} with {VirtProductName}
include::_attributes/common-attributes.adoc[]
:context: installing-oadp-kubevirt
:installing-oadp-kubevirt:
:credentials: cloud-credentials

> :provider: kubevirt

Reviewer: What is the intent of these attributes?

toc::[]

You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.

// include::snippets/oadp-mtc-operator.adoc[]

Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[{oadp-full}].

[NOTE]
====
{oadp-full} with {VirtProductName} supports the following backup and restore storage options:

* Container Storage Interface (CSI) backups

* Container Storage Interface (CSI) backups with DataMover

The following storage options are excluded:

> * File System backup and restore.
> 
> * Volume Snapshot Backups and Restores.

Reviewer: ```suggestion
* File system backup and restore

* Volume snapshot backups and restores
```

- Remove periods unless introductory phrase + list items form complete sentences. 

- Sentence case.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====

> 

Reviewer: ```suggestion
```

include::modules/install-and-configure-oadp-kubevirt.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources

> . xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[OADP plugins]

Author: ```suggestion
. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
```

> . xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
> . xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
> . xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
> .  xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]

Reviewer 2: ```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
* xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

Reviewer 2: These should be in an unordered list per the .Additional resources guidelines

> . xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
> . xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
> . xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
> . xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]

Reviewer: This must be an unordered list.

```suggestion
* xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)]
* xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR]
* xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]

> [IMPORTANT]

Reviewer: Would it be more helpful for this important admonition to be toward the top of the page?

> [IMPORTANT]

Reviewer: If this is important, why is it the last thing in the assembly? If it is something that a reader needs to know about this process, hoist it up.

Reviewer 2: +1 Max. A support statement is important and lobbing it at the bottom could lead customers down a path they can't actually take in the end.

====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups

* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.

* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====

include::modules/install-and-configure-oadp-kubevirt.adoc[leveloffset=+1]

[role="_additional-resources"]
.Additional resources

. xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[{oadp-short} plugins]
. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[Creating a `Backup` custom resource (CR)]
. xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[Creating a `Restore` CR]

> .  xref::../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]

Author: ```suggestion
.  xref::../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks]
```

:_mod-docs-content-type: ASSEMBLY
[id="installing-oadp-kubevirt"]
// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"

> = Configuring the OpenShift API for Data Protection with {VirtProductName}

Author: We would like to call it, but think reviewer will have an objection
 
```suggestion
= Configuring the OpenShift API for Data Protection with KubeVirt
```

Reviewer 3: @anarnold97 did we get any feedback on this?  IMHO my preference from first pref to last, would be.
1. OpenShift Virtualization
2. Virtualization
3. KubeVirt

Author: Not pushed it to OCP review yet. As soon as you and Amos give an LGTM, I will push it

Cool, let us go with Configuring the OpenShift API for Data Protection with {VirtProductName}

{VirtProductName} -> OpenShift Virtualization

include::_attributes/common-attributes.adoc[]
:context: installing-oadp-kubevirt
:installing-oadp-kubevirt:
:credentials: cloud-credentials
:provider: kubevirt

toc::[]

You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.

// include::snippets/oadp-mtc-operator.adoc[]

Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[OpenShift API for Data Protection (OADP)].

[NOTE]
====
(oadp-full} with {VirtProductName} backup and restore support is limited to:
* Container Storage Interface (CSI) backups
* Container Storage Interface (CSI) backups with DataMover

This support excludes:
* File System backup and restore.
* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].

> ====

Reviewer: The lists in this admonition do not render correctly in the preview. The attribute also needs to go between two brackets.

Currently:
![image](https://github.com/openshift/openshift-docs/assets/50179998/f28b054c-ffeb-48fb-9e2d-5aebe3c68b57)

This must be fixed prior to merging.

Reviewer: ```suggestion
[NOTE]
====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups
* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.
* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====
```

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

> .Procedure
> 
> . Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.
> 
> . Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
> 
> . Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].
> 
> +
> [WARNING]
> ====
> Red Hat support is limited to only the following options:
> * CSI backups
> * CSI backups with DataMover.
> ====
> 
> . Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].
> 
> To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].

Reviewer 2: ```suggestion

```

Reviewer 2: I see you're trying to do an Overview-type link situation here, but I think these kind of links need to be in == Additional resource heading like here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections

[id="configuring-dpa-kubevirt"]
== Configuring the Data Protection Application

// You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.

> include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]

Reviewer 2: I also notice that this is a bit contradictory? Its a `== Configuring`  heading but then an Installing module included? And in the preview, notice how the levels are the same? 
![image](https://github.com/openshift/openshift-docs/assets/77501157/1cb66202-12ac-4278-b1bf-ecd6bd272f1e)

Reviewer 2: I'm not sure if this is installing or configuring, but I think this area needs to be re-worked. A leveloffset=+2 applied so that the headers are at different levels and also appropriate titling for the included content.

. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

+
[WARNING]
====
Red Hat support is limited to only the following options:
* CSI backups
* CSI backups with DataMover.
====

. Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].

To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].

[id="configuring-dpa-kubevirt"]
== Configuring the Data Protection Application

// You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.

// include::modules/oadp-kubevirt-setting-resource-limits-and-requests.adoc[leveloffset=+2]

> // include::modules/oadp-self-signed-certificate.adoc[leveloffset=+2]

Author: ```suggestion
```

:_mod-docs-content-type: ASSEMBLY
[id="installing-oadp-kubevirt"]
// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"
= Configuring the OpenShift API for Data Protection with {VirtProductName}
include::_attributes/common-attributes.adoc[]
:context: installing-oadp-kubevirt
:installing-oadp-kubevirt:
:credentials: cloud-credentials

> :provider: kubevirt

Reviewer 4: why is the provider name, kubevirt? :)

Author: Do i need to change it? Nobody sees it apart from us?

Reviewer 2: If kubivert is technically not accurate, maybe it shouldn't be here? Not an expert here, but noticed this was a resolved comment that wasn't outdated and thought I'd re-raise it.

Reviewer 2: If these files are inherited by someone later, incorrect metadata could lead to confusion

toc::[]

> You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you install the Data Protection Application.

Reviewer 2: ```suggestion
You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.
```

> You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location and a snapshot location. Then, you install the Data Protection Application.

Reviewer 4: I think we can remove any reference to VSL since we only support CSI + CSI w/ DM

> You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location and a snapshot location. Then, you install the Data Protection Application.

Author: ```suggestion
You install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you install the Data Protection Application.
```

// include::snippets/oadp-mtc-operator.adoc[]

Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[OpenShift API for Data Protection (OADP)].

include::snippets/oadp-virtualization-supported-backup-types.adoc

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.

. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].

> . Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

Author: There will probably be sections added post-GA

> . Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

Author: Please can I come back next week after GA and fix this

> . Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

Reviewer 2: I checked on this with the rest of the peer review squad, and the squad agrees merging this is not okay because it does not fit with mod docs standards for assemblies: https://redhat-documentation.github.io/modular-docs/#assembly-guidelines

Assemblies can have introductions, prerequisites, and modules as the main pieces. If you have a procedure, it belongs in a procedure module. I'm not sure what other post-GA sections would validate having a procedure in an assembly 🤷‍♀️ because they just aren't allowed.

+
[WARNING]
====
Red Hat support is limited to only the following options:
* CSI backups
* CSI backups with DataMover.
====

. Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].

You create a `Secret` for the backup location and then you install the Data Protection Application. For more details, see xref:../../..//backup_and_restore/application_backup_and_restore/installing/oadp-installing-operator.adoc#oadp-installing-operator-doc[Installing the OADP Operator].

To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].

//include::modules/oadp-installing-operator.adoc[leveloffset=+1]
// include::modules/oadp-about-backup-snapshot-locations-secrets.adoc[leveloffset=+1]

> include::modules/oadp-creating-default-secret.adoc[leveloffset=+2]

Author: ```suggestion
// include::modules/oadp-creating-default-secret.adoc[leveloffset=+2]
```

> include::modules/oadp-secrets-for-different-credentials.adoc[leveloffset=+2]

Author: ```suggestion
// include::modules/oadp-secrets-for-different-credentials.adoc[leveloffset=+2]
```

include::_attributes/common-attributes.adoc[]
:context: installing-oadp-kubevirt
:installing-oadp-kubevirt:
:credentials: cloud-credentials
:provider: kubevirt

toc::[]

You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.

// include::snippets/oadp-mtc-operator.adoc[]

Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[OpenShift API for Data Protection (OADP)].

[NOTE]
====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups

* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.

* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====

> [id="install-and configure-oadp-kubevirt"]

Author: ```suggestion
[id="install-and-configure-oadp-kubevirt_context"]
```

== Installing and configuring {oadp-short} with {VirtProductName}

As a cluster administrator, you install the {oadp-short} by installing the {oadp-short} Operator. 

The Operator installs link:https://velero.io/docs/v{velero-version}[Velero velero-version].

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.

. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].

. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

+
[WARNING]
====

> Red Hat support is limited to only the following options:
> * CSI backups
> * CSI backups with DataMover.

Author: ```suggestion
Red Hat support is limited to only the following options:

* CSI backups

* CSI backups with DataMover.
```

[NOTE]
====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups

* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.

* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====

[id="install-and-configure-oadp-kubevirt_context"]
== Installing and configuring {oadp-short} with {VirtProductName}

As a cluster administrator, you install the {oadp-short} by installing the {oadp-short} Operator. 

The Operator installs link:https://velero.io/docs/v{velero-version}[Velero velero-version].

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.

> . Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].

Author: ```suggestion
. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application (DPA)] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].
```

. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

+
[WARNING]
====
Red Hat support is limited to only the following options:
* CSI backups
* CSI backups with DataMover.
====

. Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].

To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].

> [id="configuring-dpa-kubevirt"]
> == Configuring the Data Protection Application

Author: ```suggestion
```

// You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.

include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]

[IMPORTANT]
.OADP 1.2
====

> Red Hat only supports the combination of OADP versions 1.3.0 and later, and {VirtProductName} versions 4.14 and later.
> OADP versions before 1.3.0 are not supported for back up and restore of {VirtProductName}.

Author: ```suggestion
Red Hat only supports the combination of {oadp-short} versions 1.3.0 and later, and {VirtProductName} versions 4.14 and later.

{oadp-short} versions before 1.3.0 are not supported for back up and restore of {VirtProductName}.
```

:_mod-docs-content-type: ASSEMBLY
[id="installing-oadp-kubevirt"]
// If docs team approves the name, "Configuring OADP with Kubevirt" great but I suspect it will need to be "Configuring OADP with OpenShift Virtualization" or "Configuring OADP with Virtualization" or perhaps... bear with me.. "OADP and OpenShift-Virtualization"

> = Configuring the {oadp-full} API for Data Protection with {VirtProductName}

Author: ```suggestion
= Configuring the {oadp-full} with {VirtProductName}
```

include::_attributes/common-attributes.adoc[]
:context: installing-oadp-kubevirt
:installing-oadp-kubevirt:
:credentials: cloud-credentials
:provider: kubevirt

toc::[]

You can install the {oadp-first} with {VirtProductName} by installing the OADP Operator and configuring a backup location. Then, you can install the Data Protection Application.

// include::snippets/oadp-mtc-operator.adoc[]

> Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[OpenShift API for Data Protection (OADP)].

Author: ```suggestion
Back up and restore virtual machines by using the xref:../../../backup_and_restore/index.adoc#application-backup-restore-operations-overview[{oadp-full}].
```

[NOTE]
====
{oadp-full} with {VirtProductName} backup and restore support is limited to:

* Container Storage Interface (CSI) backups

* Container Storage Interface (CSI) backups with DataMover

This support excludes:

* File System backup and restore.

* Volume Snapshot Backups and Restores.

For more information, see xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-backing-up-applications-restic-doc.adoc#oadp-backing-up-applications-restic-doc[Backing up applications with File System Backup: Kopia or Restic].
====

[id="install-and-configure-oadp-kubevirt_context"]
== Installing and configuring {oadp-short} with {VirtProductName}

> As a cluster administrator, you install the {oadp-short} by installing the {oadp-short} Operator. 

Author: ```suggestion
As a cluster administrator, you install {oadp-short} by installing the {oadp-short} Operator. 
```

> The Operator installs link:https://velero.io/docs/v{velero-version}[Velero velero-version].

Author: ```suggestion
The Operator installs link:https://velero.io/docs/v{velero-version}[Velero {velero-version}].
```

.Prerequisites

* Access to the cluster as a user with the `cluster-admin` role.

.Procedure

. Install the xref:../../../backup_and_restore/application_backup_and_restore/installing/about-installing-oadp.adoc#about-installing-oadp[OADP Operator] according to the instructions for your storage provider.

. Install the xref:oadp-installing-dpa-1-3_installing-oadp-kubevirt[Data Protection Application (DPA)] with the `kubevirt` and `openshift` xref:../../../backup_and_restore/application_backup_and_restore/oadp-features-plugins.adoc#oadp-plugins_oadp-features-plugins[plugins].

. Back up virtual machines by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc#backing-up-applications[`Backup` custom resource (CR)].

+
[WARNING]
====
Red Hat support is limited to only the following options:

* CSI backups

* CSI backups with DataMover.
====

. Restore the `Backup` CR by creating a xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/restoring-applications.adoc#restoring-applications[`Restore` CR].

> To install the OADP Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].

Author: ```suggestion
To install the {oadp-short} Operator in a restricted network environment, you must first disable the default OperatorHub sources and mirror the Operator catalog. For details, see xref:../../../operators/admin/olm-restricted-networks.adoc#olm-restricted-networks[Using Operator Lifecycle Manager on restricted networks].
```

// You can configure the Data Protection Application by setting Velero resource allocations or enabling self-signed CA certificates.

include::modules/oadp-installing-dpa-1-3.adoc[leveloffset=+1]

[IMPORTANT]

> .OADP 1.2

Author: ```suggestion
.{oadp-short} 1.2
```
